src/core/MultiFormatReader.ts

```diff
@@ -101,41 +101,42 @@ export default class MultiFormatReader implements Reader {
 
         const tryHarder: boolean = hints !== null && hints !== undefined && undefined !== hints.get(DecodeHintType.TRY_HARDER);
         /*@SuppressWarnings("unchecked")*/
-        const formats = hints === null || hints === undefined ? null : hints.get(DecodeHintType.POSSIBLE_FORMATS);
+        const formats = hints === null || hints === undefined ? null : <BarcodeFormat[]>hints.get(DecodeHintType.POSSIBLE_FORMATS);
         const readers = new Array<Reader>();
         if (formats !== null && formats !== undefined) {
-            const addOneDReader: boolean =
-                formats.contains(BarcodeFormat.UPC_A) ||
-                formats.contains(BarcodeFormat.UPC_E) ||
-                formats.contains(BarcodeFormat.EAN_13) ||
-                formats.contains(BarcodeFormat.EAN_8) ||
-                formats.contains(BarcodeFormat.CODABAR) ||
-                formats.contains(BarcodeFormat.CODE_39) ||
-                formats.contains(BarcodeFormat.CODE_93) ||
-                formats.contains(BarcodeFormat.CODE_128) ||
-                formats.contains(BarcodeFormat.ITF) ||
-                formats.contains(BarcodeFormat.RSS_14) ||
-                formats.contains(BarcodeFormat.RSS_EXPANDED);
+            const addOneDReader: boolean = formats.some(f =>
+                f === BarcodeFormat.UPC_A ||
+                f === BarcodeFormat.UPC_E ||
+                f === BarcodeFormat.EAN_13 ||
+                f === BarcodeFormat.EAN_8 ||
+                f === BarcodeFormat.CODABAR ||
+                f === BarcodeFormat.CODE_39 ||
+                f === BarcodeFormat.CODE_93 ||
+                f === BarcodeFormat.CODE_128 ||
+                f === BarcodeFormat.ITF ||
+                f === BarcodeFormat.RSS_14 ||
+                f === BarcodeFormat.RSS_EXPANDED
+            );
             // Put 1D readers upfront in "normal" mode
 
             // TYPESCRIPTPORT: TODO: uncomment below as they are ported
 
             if (addOneDReader && !tryHarder) {
                readers.push(new MultiFormatOneDReader(hints));
             }
-            if (formats.contains(BarcodeFormat.QR_CODE)) {
+            if (formats.indexOf(BarcodeFormat.QR_CODE) >= 0) {
```

Comment:
There is no `contains` in Typescript, as far as I know. At least it didn't compile.

Reply:
Ohhh, of course! What about (`includes`)[https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/includes]?

Reply:
It didn't compile, neither. I think it would be necessary to update Typescript version and/or use a polyfill.

Reply:
I never had a problem with contains, maybe it's your typescript or something else that is causing the issue. Also, TravisCI builds never broke because of `contains` use. That puts me in a complicated position. 🤔

Reply:
Well, found the problem, we'r missing the `"es7",` key on `"lib"` configuration inside `tsconfig.json`. Shall we add it? 😃